sentry-core/src/session.rs

```diff
@@ -0,0 +1,316 @@
+//! Release Health Sessions
+//!
+//! https://develop.sentry.dev/sdk/sessions/
+
+use std::borrow::Cow;
+use std::sync::Arc;
+use std::time::Instant;
+
+use crate::protocol::{Event, Level, User};
+use crate::scope::StackLayer;
+use crate::types::{DateTime, Utc, Uuid};
+
+/// Represents the status of a session.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum SessionStatus {
+    Ok,
+    Crashed,
+    #[allow(dead_code)]
+    Abnormal,
+    Exited,
+}
+
+pub enum SessionUpdate {
+    NeedsFlushing(Session),
+    Unchanged,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Session {
+    session_id: Uuid,
+    status: SessionStatus,
+    errors: usize,
+    user: Option<Arc<User>>,
+    release: Cow<'static, str>,
+    environment: Option<Cow<'static, str>>,
+    started: Instant,
+    started_utc: DateTime<Utc>,
+    duration: Option<f64>,
```

Comment:
I had to look up the api docs to find out this is "seconds as float".  Either documenting this or just keeping it here as a `Duration` until it's serialised might be better?  I think I'd vote for `Duration`.